x-pack/test/api_integration/apis/ml/data_frame_analytics/get_spaces.ts

```diff
@@ -140,5 +153,45 @@ export default ({ getService }: FtrProviderContext) => {
     it('should fail to list jobs stats by job ids if one of them is in a different space', async () => {
       await runRequest(idSpace1, 404, true, `${jobIdSpace1},${jobIdSpace2}`);
     });
+
+    describe('GetDataFrameAnalyticsIdMap with spaces', () => {
+      it(`should return a map of objects from ${idSpace1} leading up to analytics job id created in ${idSpace1}`, async () => {
+        const body = await runMapRequest(idSpace1, 200, jobIdSpace1);
+
+        expect(body).to.have.keys('elements', 'details', 'error');
+        // Index node, 2 job nodes (with same source index), and 2 edge nodes to connect them
+        expect(body.elements.length).to.eql(5);
+        // No space2 related job ids should be returned
+        for (const detailsId in body.details) {
+          if (detailsId.includes('analytics')) {
+            expect(body.details[detailsId].id.includes(idSpace2)).to.eql(false);
```

Comment:
Would be good to add an assertion message, otherwise a failure would e.g. look like `expected true to kind of equal false`. Could be something like this:
```suggestion
            expect(body.details[detailsId].id.includes(idSpace2)).to.eql(false, `No space2 related job ids should be returned, got ${body.details[detailsId].id}`);
```

Reply:
Added in 08bde443ff73808696c65900e93c6bf22fa70265